src/theory/builtin/type_enumerator.h

```diff
@@ -75,6 +76,49 @@ class UninterpretedSortEnumerator : public TypeEnumeratorBase<UninterpretedSortE
 
 };/* class UninterpretedSortEnumerator */
 
+
```

Comment:
Give this its own header & c++ file

Reply:
I think this is overkill.  The precedent we've had is to group type rules / type enumerators into a single file per theory.

Reply:
Not sure I agree, but adding the cpp is already a big step forward so lets proceed.